Src/Newtonsoft.Json/JsonReader.Async.cs

```diff
@@ -105,6 +107,27 @@ public virtual Task<byte[]> ReadAsBytesAsync(CancellationToken cancellationToken
             return cancellationToken.CancelIfRequestedAsync<byte[]>() ?? Task.FromResult(ReadAsBytes());
         }
 
+
+        internal async Task<byte[]> ReadArrayIntoByteArrayAsync(CancellationToken cancellationToken)
+        {
+            List<byte> buffer = new List<byte>();
+
+            for (;;)
```

Comment:
Yeah, they're exactly the same. To me `for (;;)` reads as "forever" and `while (true)` reads as "while the basic principles of Aristotelian logic and Boolean logic still hold", and hence a strange way to express "forever", but I think that might be a very personal take on which to use 😄